app/admin.go

```diff
@@ -62,20 +58,51 @@ func (a *App) GetLogsSkipSend(page, perPage int) ([]string, *model.AppError) {
 
 		defer file.Close()
 
-		offsetCount := 0
-		limitCount := 0
-		scanner := bufio.NewScanner(file)
-		for scanner.Scan() {
-			if limitCount >= perPage {
-				break
+		stat, err := file.Stat()
+		if err != nil {
+			return nil, model.NewAppError("getLogs", "api.admin.file_read_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+
+		var newLine = []byte{'\n'}
+		var lineCount int
+		var searchPos = int64(-1)
```

Comment:
You never set `searchPos` to anything other than `-1`. Is this intentional?

Reply:
Yes, I use `searchPos` for clarifying that log file are being read in reverse order.

As a proposal, How about deleting `searchPos` and adding comments `// Reading log file in reverse order`.